cmd/agent/app/agent.go

```diff
@@ -132,6 +132,12 @@ func run(ctx context.Context, karmadaConfig karmadactl.KarmadaConfig, opts *opti
 		ControlPlaneConfig: controlPlaneRestConfig,
 		ClusterConfig:      clusterConfig,
 	}
+	ns, err := clusterKubeClient.CoreV1().Namespaces().Get(ctx, "kube-system", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	registerOption.ClusterID = string(ns.UID)
```

Comment:
I'm not sure this is a reasonable change,
https://github.com/karmada-io/karmada/blob/ec84d649a9975435f58f690493e943204f9fbff6/cmd/agent/app/agent.go#L135-L140
https://github.com/karmada-io/karmada/blob/ec84d649a9975435f58f690493e943204f9fbff6/pkg/karmadactl/join.go#L180-L189
This both can also move into RegisterClusterInControllerPlane.But I don't know which will be better.
The function RegisterClusterInControllerPlane and others' ability is not very clear.

Reply:
I think `RegisterClusterInControllerPlane` can do all thing that needs to be done for registering the cluster so that the login in the current `run` function will be clear. This cleanup may need to be done in a separate pr.

Reply:
clusterKubeClient is created by clusterConfig but RegisterClusterInControllerPlane don't have clusterConfig.So it will result in a bigger change if do so.